app/src/main/java/org/kiwix/kiwixmobile/utils/KiwixDialog.kt

```diff
@@ -30,31 +32,66 @@ sealed class KiwixDialog(
   }
 
   object LocationPermissionRationale : KiwixDialog(
-      null, R.string.permission_rationale_location, android.R.string.yes, android.R.string.cancel
+    null,
+    R.string.permission_rationale_location,
+    android.R.string.yes,
+    android.R.string.cancel,
+    null
   )
 
   object StoragePermissionRationale : KiwixDialog(
-      null, R.string.request_storage, android.R.string.yes, android.R.string.cancel
+    null, R.string.request_storage, android.R.string.yes, android.R.string.cancel, null
   )
 
   object EnableWifiP2pServices : KiwixDialog(
-      null, R.string.request_enable_wifi, R.string.yes, android.R.string.no
+    null, R.string.request_enable_wifi, R.string.yes, android.R.string.no, null
   )
 
   object EnableLocationServices : KiwixDialog(
-      null, R.string.request_enable_location, R.string.yes, android.R.string.no
+    null, R.string.request_enable_location, R.string.yes, android.R.string.no, null
+  )
+
+  object TurnOffHotspotManually : KiwixDialog(
+    R.string.hotspot_failed_title,
+    R.string.hotspot_failed_message,
+    R.string.go_to_wifi_settings_label,
+    null,
+    null
+  )
+
+  data class ShowHotspotDetails(override val args: Array<out Any>) : KiwixDialog(
+    R.string.hotspot_turned_on,
+    R.string.hotspot_details_message,
+    android.R.string.ok,
+    null,
+    null
+  ), HasBodyFormatArgs {
+    constructor(wifiConfiguration: WifiConfiguration) : this(
+      arrayOf(
+        wifiConfiguration.SSID,
+        wifiConfiguration.preSharedKey
+      )
+    )
+  }
+
+  object StartHotspotManually : KiwixDialog(
```

Comment:
StartHotspotManually (I believe in example code I gave you) was of the form
```
data class StartHotspotManually(val neutralMessage:Int=R.string.your_netral_message):KiwixDialog(R.string.your_title….)
```
Actually object is a slightly better way to go but to add some extra data we can't supply it to a constructor like we are doing there.

```
object StartHotspotManually : KiwixDialog(R.string.your_title….){
   val neutralMessage = R.string.your_neutral_message
}

Reply:
Fixed StartHotspotManually